wallet/chainntfns.go

```diff
@@ -47,42 +52,45 @@ func (w *Wallet) handleChainNotifications() {
 		// if it doesn't match the original hash returned by
 		// the notification, to roll back and restart the
 		// rescan.
-		log.Infof("Catching up block hashes to height %d, this"+
-			" might take a while", height)
-		err := walletdb.Update(w.db, func(tx walletdb.ReadWriteTx) error {
-			ns := tx.ReadWriteBucket(waddrmgrNamespaceKey)
-
-			startBlock := w.Manager.SyncedTo()
+		log.Infof("Catching up block hashes to height %d, this might "+
+			"take a while", height)
+
+		heightRange := &scanRange{start: startHeight, end: height}
+		var blockStamps []*waddrmgr.BlockStamp
+		err := scanChain(chainClient, heightRange, func(height int32,
+			hash *chainhash.Hash, header *wire.BlockHeader) error {
+
+			blockStamps = append(blockStamps, &waddrmgr.BlockStamp{
+				Hash:      *hash,
+				Height:    height,
+				Timestamp: header.Timestamp,
+			})
+			return nil
 
-			for i := startBlock.Height + 1; i <= height; i++ {
-				hash, err := client.GetBlockHash(int64(i))
```

Comment:
whoa, we were doing RPC calls withing a DB tx? 😮

Reply:
Yeah we do this in other places too, like the recovery logic.